core/src/main/java/it/pagopa/selfcare/mscore/core/DelegationServiceImpl.java

```diff
@@ -40,16 +43,23 @@ public Delegation createDelegation(Delegation delegation) {
         }
         if (PROD_PAGOPA.equals(delegation.getProductId())) {
             List<Institution> institutions = institutionService.getInstitutions(delegation.getTo(), null);
-            String partnerIdentifier = institutions.stream()
+            Institution partner = institutions.stream()
                     .findFirst()
-                    .map(Institution::getId)
                     .orElseThrow(() -> new ResourceNotFoundException(String.format(INSTITUTION_TAX_CODE_NOT_FOUND.getMessage(), delegation.getTo()),
                             INSTITUTION_TAX_CODE_NOT_FOUND.getCode()));
-            delegation.setTo(partnerIdentifier);
+            delegation.setTo(partner.getId());
         }
         try {
+            delegation.setCreatedAt(OffsetDateTime.now());
+            delegation.setUpdatedAt(OffsetDateTime.now());
+            delegation.setStatus(DelegationState.ACTIVE);
             Delegation savedDelegation = delegationConnector.save(delegation);
-            notificationService.sendMailForDelegation(delegation.getInstitutionFromName(), delegation.getProductId(), delegation.getTo());
+            institutionService.updateInstitutionDelegation(delegation.getTo(), true);
+            try {
+                notificationService.sendMailForDelegation(delegation.getInstitutionFromName(), delegation.getProductId(), delegation.getTo());
+            } catch (Exception e) {
+                throw new MsCoreException(SEND_MAIL_FOR_DELEGATION_ERROR.getMessage(), SEND_MAIL_FOR_DELEGATION_ERROR.getCode());
+            }
```

Comment:
it should be out of the first try/catch otherwise MsCoreException will be catched....you could put them sequentially

Reply:
resolved